impress-org/givewp-next-gen
# Feature: Add Reveal and Modal display options to Donation Form Block v3

## Description

This PR updates the current Donation Form Block v(3) with an additional block control named ```Form Format``` which is a dropdown that allows users to choose from two new display options to render their donation form. 

The two new options are "Reveal" & "Modal". Both display options are compatible with all 3 form templates. Both display options initially hide the users form & only render a single button. If Either of these options are selected an additional control to edit the text within this button becomes available.

If "Reveal" is selected, the form is hidden and when the user clicks the button the form is "revealed" just beneath the button.

If "Modal" is selected, the form is hidden and when the user clicks the button the form is rendered in a modal in the middle of the screen.


## Affects
Donation Form Block v3

## Visuals

https://www.loom.com/share/3b3e92d800e148e5ae06a7dc9e3f4419?sid=ea708d9c-6752-476c-b60d-d907eb7823cb

## Testing Instructions

- Create a v3 form for each form template.
- Create a page and add the Donation Form Block v3 for each template you created.
- Select the block and set the display option to "Reveal" on all three forms.
- Test the additional text control for the 'Open form button'
- While in the block editor verify the forms are hidden and reveal when the button is clicked.
- View the page and verify again that it works properly on the frontend page.
- Repeat steps for the modal option.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204982550616184
